src/app/clusters/level-control/level-control.cpp

```diff
@@ -580,6 +581,28 @@ static void setOnOffValue(EndpointId endpoint, bool onOff)
 #endif // MATTER_DM_PLUGIN_ON_OFF
 }
 
+static Status VerifyOptionsBitmap(Optional<BitMask<OptionsBitmap>> optionsMask, Optional<BitMask<OptionsBitmap>> optionsOverride)
+{
+    constexpr BitMask<OptionsBitmap> maxOptionValue(
+        OptionsBitmap::kExecuteIfOff, OptionsBitmap::kCoupleColorTempToLevel); // add any new bitmap field to this value!
+
+    // Based on spec, we shall expect a value for those arguments. But that's not what has been done in SDK so far.
+    // Should we error out?
+    if (optionsMask.HasValue())
+    {
+        VerifyOrReturnError(optionsMask.Value() <= maxOptionValue, Status::ConstraintError,
+                            ChipLogError(Zcl, "Command error: out of range optionsMask argument"));
```

Comment:
Is this the behavior the spec calls for?

Reply:
@bzbarsky-apple I did not find any indication in the cluster spec that specifies what to do in case of unknown bits options bitfield. 
My assumption was to treat them as values outside of the constraint of the expected parameter, like other argument types.

Ticket #34554 complains about unrespected constraints after some fuzz testing found those issues, so I was trying to address that.

However in the main spec, says that Reserved bit/undefined by conveyed from another implementation is ignored

> 1.8.2. Reserved Bit Fields
> An undefined bit or bit field SHALL be considered reserved for future revisions of this standard and SHALL NOT be available for implementation.
> 
> It is RECOMMENDED that a bit stay undefined, rather than defining it as "reserved".
> 
> An implementation of a revision where a bit is reserved SHALL indicate that bit as zero when conveying that bit in an interaction, **and ignore that bit when conveyed from another implementation.**

I should close this PR then and reply, in the issue ticket, that it is an expected behaviour. 

Does that work for you?

Reply:
The basic question here is: what should happen when a newer-revision client (which knows about more bits) talks to an older-revision server?  Just ignoring the unknown bits would make sense to me, but really the lighting spec people should figure this out.